Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -485,17 +485,25 @@ public typealias StartPurchaseBlock = (@escaping PurchaseCompletedBlock) -> Void
         let notificationCenter: NotificationCenter = .default
         let purchasesOrchestrator: PurchasesOrchestrator = {
             if #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) {
-                var diagnosticsSynchronizer: DiagnosticsSynchronizer?
+                let diagnosticsSynchronizer: DiagnosticsSynchronizer?
                 if diagnosticsEnabled {
                     if let diagnosticsFileHandler = diagnosticsFileHandler {
                         let synchronizedUserDefaults = SynchronizedUserDefaults(userDefaults: userDefaults)
                         diagnosticsSynchronizer = DiagnosticsSynchronizer(internalAPI: backend.internalAPI,
                                                                           handler: diagnosticsFileHandler,
                                                                           tracker: diagnosticsTracker,
                                                                           userDefaults: synchronizedUserDefaults)
+                        if let diagnosticsFileHandler = diagnosticsFileHandler as? DiagnosticsFileHandler {
```

Comment:
I think that the casting is not necessary because the `updateDelegate` method is part of the `DiagnosticsFileHandlerType` protocol.
```suggestion
                        if let diagnosticsFileHandler = diagnosticsFileHandler {
```

Reply:
Right! This is from a previous iteration, thanks! I belive I can remove the entire `if let`

Reply:
Yeah, much better even. Thanks!